### PR TITLE
Add IT for interface

### PR DESCRIPTION
**Pull request by b4sjoo**
### Description
Add IT for interface

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 1]
```diff
@@ -237,6 +237,42 @@ public void testPredictRemoteModel() throws IOException, InterruptedException {
...
+        String taskId = (String) responseMap.get("task_id");
+        waitForTask(taskId, MLTaskState.COMPLETED);
+        response = getTask(taskId);
+        responseMap = parseResponseToMap(response);
+        String modelId = (String) responseMap.get("model_id");
+        response = deployRemoteModel(modelId);
+        responseMap = parseResponseToMap(response);
+        taskId = (String) responseMap.get("task_id");
+        waitForTask(taskId, MLTaskState.COMPLETED);
+        String predictInput = "{\n" + "  \"parameters\": {\n" + "      \"prompt\": \"Say this is a test\"\n" + "  }\n" + "}";
```
Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:259]: Can you add a test case for a wrong input to verify if it does fail ?

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 2]
```diff
@@ -237,6 +237,70 @@ public void testPredictRemoteModel() throws IOException, InterruptedException {
...
+        try {
+            response = predictRemoteModel(modelId, predictInput);
+            responseMap = parseResponseToMap(response);
+            verifyResponse.accept(responseMap);
+        } catch (Exception e) {
+            verifyException.accept(e);
```
Comment (mingshl) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:267]: wondering if can check on the model interface in the response?

Comment (b4sjoo) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:267]: What do you mean by "check on the model interface in the response"

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java (hunk 1 above)
Comment (b4sjoo) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:261]: Added

File: ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java [hunk 3]
```diff
@@ -220,7 +220,7 @@ public void invokeRemoteModel_nullHttpClient_throwMLException() throws NoSuchFie
                 new ExecutionContext(0, new CountDownLatch(1), new AtomicReference<>()),
                 actionListener
             );
-        ArgumentCaptor<Exception> argumentCaptor = ArgumentCaptor.forClass(Exception.class);
+        ArgumentCaptor<Exception> argumentCaptor = ArgumentCaptor.forClass(NullPointerException.class);
```
Comment (ylwu-amzn) [inline ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java:223]: Why change to NullPointerException?

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java (hunk 2 above)
Comment (mingshl) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:267]: you can try verify the response for connector_id and model_id

https://github.com/opensearch-project/ml-commons/blob/8cb6a9b6de485f57bf0b3a8e8ea9905aa292ad0f/plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java#L633

File: ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java (hunk 3 above)
Comment (b4sjoo) [reply ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java:223]: This will fail the UT

Comment (b4sjoo) [reply ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java:223]: I don't know why, it's just failed, https://github.com/opensearch-project/ml-commons/actions/runs/8915168818/job/24484230861

Comment (b4sjoo) [reply ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java]: Seems this UT is flaky, I reverted the change.

**Review by ylwu-amzn — APPROVED** [review]

**Review by mingshl — APPROVED** [review]

**Pull request merged**